### PR TITLE
support dev dependencies

### PR DESCRIPTION
**Pull request by ccorcos**
- properly using defaults
- added support for parsing dev dependencies using `--include devDependencies`

### CODE REVIEW

File: lib/args.js [hunk 1]
```diff
@@ -24,13 +25,17 @@ var nopt = require('nopt'),
         "json" : "Path to be scanned",
         "version" : "Show the current version",
         "export" : "Export format, 'tree' or 'json'",
-        "help" : "This help"
+        "help" : "This help",
+        "include": "Set this to dependencies or devDependencies",
```
Comment (badunk) [inline lib/args.js]: There are [peer dependencies](https://docs.npmjs.com/files/package.json#peerdependencies), bundled dependecies and optional dependencies too.

File: lib/index.js [hunk 2]
```diff
@@ -113,7 +113,8 @@ exports.init = function(options, callback) {
     }
     
     var opt = {
-        depth: options.depth
+        depth: options.depth,
+        dev: options.include.indexOf('devDependencies') != -1,
```
Comment (badunk) [inline lib/index.js]: does `dev`: actually do anything?  maybe I missed it but I don't think its read anywhere

**Comment by badunk** [comment]
dev dependencies are supported in general, but perhaps only when used as an npm package.  Is the problem that it isn't able to be filtered out in the command line?

File: lib/args.js (hunk 1 above)
Comment (ccorcos) [inline lib/args.js]: true

File: lib/index.js (hunk 2 above)
Comment (ccorcos) [inline lib/index.js]: yes, I looked at the documentation for [`read-installed`](https://github.com/npm/read-installed), and it has a dev option. Not options for bundled or optional dependencies...

**Comment by ccorcos** [comment]
I added the option to the command line AND fixed it for both requiring through node and the cli.

**Comment by ccorcos** [comment]
@badunk currently the dev dependencies arent parsed at all because you need to pass a `dev=true` options to [read-installed](https://github.com/npm/read-installed). Are you going to accept this fix?

**Comment by badunk** [comment]
I think I finally understand what you're saying

Thanks!

**Comment by ccorcos** [comment]
Awesome. Let me know when you update the version to I can change it on my end

**Comment by badunk** [comment]
already pushed, give it a try

**Comment by ccorcos** [comment]
Awesome. Thanks!

**Pull request merged**